lib/xebow_web/live/matrix_live.ex

```diff
@@ -0,0 +1,191 @@
+defmodule XebowWeb.MatrixLive do
+  use XebowWeb, :live_view
+
+  alias RGBMatrix.{Animation, Engine}
+
+  @layout Xebow.layout()
+  @black Chameleon.HSV.new(0, 0, 0)
+  @black_frame @layout
+               |> Layout.leds()
+               |> Map.new(fn led ->
+                 {led.id, @black}
+               end)
+
+  @impl Phoenix.LiveView
+  def mount(_params, _session, socket) do
+    if connected?(socket) do
+      register_with_engine!()
+    end
+
+    {config, config_schema} = Engine.get_animation_config()
```

Comment:
:information_source: 

Tying this back to one of my prior comments, this feels like animations are being coupled to the engine when they probably shouldn't be. This feels more like wanting to get information from the application. How does this subtle change in naming feel when making this function call?

```ex
{config, config_schema} =
  Xebow.get_current_animation
  |> Animation.get_config
```

Reply:
But wouldn't the implementation of `Xebow.get_current_animation` need to reach into the engine state to get the current animation struct anyway?

Animations are an RGBMatrix concern, not a Xebow concern right?

I think we're having some confusion over what owns what, let's have a realtime conversation about this.

Reply:
>But wouldn't the implementation of `Xebow.get_current_animation` need to reach into the engine state to get the current animation struct anyway?

What I'm trying to point out with this name is that the engine doesn't have to be responsible for providing public information about the current animation. The application layer (`Xebow` in this case) could do that.

>I think we're having some confusion over what owns what, let's have a realtime conversation about this.

Definitely. We should probably work out a new architecture diagram for what the system should look like as part of that discussion.

Reply:
Oh, I've also got a crazy idea that involves storing the animations, or more specifically their config, in an ETS table if we need a better model of shared access to them (wrap the ETS access with helper functions, like in `Animation`, to expose a clean public interface). I _think_ ETS tables are good at shared read access but we need to validate that, because I also hear people tend to create bottlenecks around ETS and we don't want to fall into that trap.

Reply:
Indeed, http://erlang.org/doc/man/ets.html#new_2_read_concurrency

Reply:
I think I'm starting to see what you're pushing for, but there's still a big problem.  The animation struct `Animation.t()` is what holds the state, which is constantly updating, 60 times per second for some animations.  The engine HAS to the the thing that holds on to that struct and passes it in and receives it back from the animation implementation.

If you want the Xebow application to be responsible for which animation is the currently selected one and what its current config it, then we may need to split this struct up into two: `Animation.t()` which can be owned by Xebow and `ActiveAnimation.t()` which can hold the current state and is owned by Engine. But this starts feeling a bit weird to me, because then they're not linked and can theoretically become out of sync.  Xebow could lose track of which animation is actually being currently "run" by the engine...

It truly feels to me that Engine 100% owns the current animation, it's state, and it's configuration.